llvm/include/llvm/ADT/STLExtras.h

```diff
@@ -1694,6 +1695,11 @@ template <typename R> constexpr size_t range_size(R &&Range) {
     return static_cast<size_t>(std::distance(adl_begin(Range), adl_end(Range)));
 }
 
+/// Wrapper for std::accumulate.
+template <typename R, typename E> E range_accumulate(R &&Range, E Init) {
+  std::accumulate(Range.begin(), Range.end(), Init);
```

Comment:
we could perfect-forward `Init` and save one copy even though the STL function takes it by value